pkg/route/generator/generate.go

```diff
@@ -77,8 +78,14 @@ func (RouteGenerator) Generate(genericParams map[string]interface{}) (runtime.Ob
 
 	portString := params["target-port"]
 	if len(portString) > 0 {
+		var targetPort intstr.IntOrString
+		if port, err := strconv.Atoi(portString); err == nil {
+			targetPort = intstr.FromInt(port)
+		} else {
+			targetPort = intstr.FromString(portString)
+		}
 		route.Spec.Port = &api.RoutePort{
-			TargetPort: intstr.FromString(portString),
+			TargetPort: targetPort,
```

Comment:
It would work but since route port is a IntOrString, I would prefer to save the correct type in it (inststr.Int in case of an int, intstr.String in case of a string) and not go with [string type](https://github.com/kubernetes/kubernetes/blob/f046d6c83e82d48e5f32539bae677661f3c505c4/pkg/util/intstr/intstr.go#L61) all the time.

The code in the oc status marker is deliberately like that to handle cases where we have saved int types as strings (what this code was doing before) but it is wrong, right?